src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacKeyFieldDeclaration/KeyFieldDeclarationFix.cs

```diff
@@ -0,0 +1,134 @@
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Acuminator.Analyzers.StaticAnalysis.DacKeyFieldDeclaration
+{
+	[Shared]
+	[ExportCodeFixProvider(LanguageNames.CSharp)]
+	public class KeyFieldDeclarationFix : CodeFixProvider
+	{
+		private const string IsKey = nameof(PX.Data.PXDBFieldAttribute.IsKey);
+		
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+			ImmutableArray.Create(Descriptors.PX1055_DacKeyFieldBound.Id);
+
+		public override  Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			return Task.Run(() =>
+			{
+				var diagnostic = context.Diagnostics.FirstOrDefault(d => d.Id == Descriptors.PX1055_DacKeyFieldBound.Id);
+
+				context.CancellationToken.ThrowIfCancellationRequested();
+
+				if (diagnostic == null)
+					return;
+
+				string codeActionIdentityKeyName = nameof(Resources.PX1055Fix1).GetLocalized().ToString();
+				CodeAction codeActionIdentityKey = CodeAction.Create(codeActionIdentityKeyName,
+																	cToken => RemoveKeysFromFieldsAsync(context.Document,
+																										context,
+																										cToken, 
+																										diagnostic, 
+																										editIdentityAttribute:false),
+																	equivalenceKey: codeActionIdentityKeyName);
+
+				string codeActionBoundKeysName = nameof(Resources.PX1055Fix2).GetLocalized().ToString();
+				CodeAction codeActionBoundKeys = CodeAction.Create(	codeActionBoundKeysName,
+																	cToken => RemoveKeysFromFieldsAsync(context.Document, 
+																										context,
+																										cToken, 
+																										diagnostic, 
+																										editIdentityAttribute: true),
+																	equivalenceKey: codeActionBoundKeysName);
+
+				string codeActionRemoveIdentityColumnName = nameof(Resources.PX1055Fix3).GetLocalized().ToString();
+				CodeAction codeActionRemoveIdentityColumn = CodeAction.Create(codeActionRemoveIdentityColumnName,
+																				cToken => RemoveKeysFromFieldsAsync(context.Document, 
+																													context, 
+																													cToken, 
+																													diagnostic,
+																													removeIdentityAttribute: true),
+																				equivalenceKey: codeActionRemoveIdentityColumnName);
+
+				context.RegisterCodeFix(codeActionIdentityKey, context.Diagnostics);
+				context.RegisterCodeFix(codeActionBoundKeys, context.Diagnostics);
+				context.RegisterCodeFix(codeActionRemoveIdentityColumn, context.Diagnostics);
+			}, context.CancellationToken);
+			
+		}
+
+		private async Task<Document> RemoveKeysFromFieldsAsync(Document document,CodeFixContext context, CancellationToken cToken, Diagnostic diagnostic, bool editIdentityAttribute = false, bool removeIdentityAttribute = false)
```

Comment:
spacing; commonly used parameter name for `CancellationToken` is `cancellationToken`